examples/agent_services_example.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+__copyright__ = 'Copyright 2013-2014, http://radical.rutgers.edu'
+__license__   = 'MIT'
+
+import os
+import sys
+
+import radical.pilot as rp
+import radical.utils as ru
+
+if __name__ == '__main__':
+
+    # we use a reporter class for nicer output
+    report = ru.Reporter(name='radical.pilot')
+    report.title('Getting Started (RP version %s)' % rp.version)
+
+    # use the resource specified as argument, fall back to localhost
+    if   len(sys.argv)  > 2: report.exit('Usage:\t%s [resource]\n\n' % sys.argv[0])
+    elif len(sys.argv) == 2: resource = sys.argv[1]
+    else                   : resource = 'local.localhost'
+
+
+    # Create a new session. No need to try/except this: if session creation
+    # fails, there is not much we can do anyways...
+    session = rp.Session()
+
+    # all other pilot code is now tried/excepted.  If an exception is caught, we
+    # can rely on the session object to exist and be valid, and we can thus tear
+    # the whole RP stack down via a 'session.close()' call in the 'finally'
+    # clause...
+    try:
+
+        # read the config used for resource details
+        config = ru.read_json('%s/config.json'
+                            % os.path.dirname(__file__)).get(resource, {})
+        pmgr   = rp.PilotManager(session=session)
+        tmgr   = rp.TaskManager(session=session)
+
+        report.header('submit pilots')
+
+        # Add a PilotManager. PilotManagers manage one or more pilots.
+
+        # Define an [n]-core local pilot that runs for [x] minutes
+        # Here we use a dict to initialize the description object
+        pd_init = {'resource'      : resource,
+                   'runtime'       : 30,  # pilot runtime (min)
+                   'exit_on_error' : True,
+                   'project'       : config.get('project'),
+                   'queue'         : config.get('queue'),
+                   'access_schema' : config.get('schema'),
+                   'cores'         : config.get('cores', 1),
+                   'gpus'          : config.get('gpus',  0),
+                   # TODO create shell script
+                   'services'       :[rp.TaskDescription({'executable':'free -h'}),rp.TaskDescription({'executable':'free -h'}) ]
```

Comment:
max line length: 80 chars

Reply:
Fixed

Reply:
Changed, not fixed - still longer than 80 chars (some other lines in the file, too).

Reply:
Yes. I have now configure hard wrap at 80. So this will never occur again.